sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -441,6 +441,28 @@ object ScalaReflection extends ScalaReflection {
           val newPath = s"""- array element class: "$clsName"""" +: walkedTypePath
           MapObjects(serializerFor(_, elementType, newPath), input, dt)
 
+         case dt @ (BooleanType | ByteType | ShortType | IntegerType | LongType |
```

Comment:
I am afraid that I cannot understand your question correctly.
This case covers only cases that we are interested in (e.g. generate `UnsafeArrayData`). Other cases are covered by [here](https://github.com/apache/spark/pull/15044/files/df300c534670e65cb44ae570859711382f85043b#diff-89643554d9757dd3e91abff1cc6096c7R466).

Is this an answer to you?